src/public/javascripts/testingtool.js

```diff
@@ -1,202 +1,202 @@
-  });
-}
-
-/**
- * Fills all tables with data from a specific time
- * @param {Number} time the time the data is located
- */
-function fillAllTables(time) {
-  let subsystems = Object.keys(runData);
-  subsystems.forEach((system) => {
-    fillTable(`${system}`, time);
-  });
-}
-
-/**
- * If there is run data fill all tables and update labels
- */
-function updateTables() {
-  if (runData) { fillAllTables(currentTime); }
-  TIME_LABEL.innerHTML = `Current Time: ${currentTime} Frames | ${currentTime / 10} Seconds`;
-  MAX_TIME_LABEL.innerHTML = `Total Time: ${runMax} Frames | ${runMax / 10} Seconds`;
-}
-setInterval(updateTables, 100);
-
-// Event Listeners
-FORWARD_BUTTON.addEventListener('click', () => { currentTime += 1; });
-BACKWARDS_BUTTON.addEventListener('click', () => { currentTime -= 1; });
-RESET_BUTTON.addEventListener('click', resetTimer);
-PLAY_BUTTON.addEventListener('click', playTimer);
-PAUSE_BUTTON.addEventListener('click', pauseTimer);
-FILE_INPUT.addEventListener('change', handleFiles, false);
-FILE_SELECTOR.addEventListener('click', () => { FILE_INPUT.click(); });
-
-// Init
-/**
- * Fills all tables and dropdowns
- */
-function init() {
-  DYNAMIC_LOADING.fillAllTables();
-  DYNAMIC_LOADING.fillAllItems(true);
-}
-init();
-// End Init
+/**
+ * @module TestingTool
+ * @author Eric Udlis
+ * @description The backend of the testing tool
+ */
+const FS = require('fs');
+const DYNAMIC_LOADING = require('./public/javascripts/dynamicloading');
+
+const FILE_SELECTOR = document.getElementById('fileSelector');
+const FILE_INPUT = document.getElementById('file-input');
+const FORWARD_BUTTON = document.getElementById('forward');
+const BACKWARDS_BUTTON = document.getElementById('back');
+const TIME_LABEL = document.getElementById('currentTime');
+const MAX_TIME_LABEL = document.getElementById('maxTime');
+const PLAY_BUTTON = document.getElementById('play');
+const PAUSE_BUTTON = document.getElementById('pause');
+const RESET_BUTTON = document.getElementById('reset');
+const SCRUBBER = document.getElementById('scrubber');
+let runData = null;
+let runMax = 'No File Loaded';
+let maxReached = false;
+let currentTime = 0;
+let timer;
+let isPlaying = false;
+
+
+// Timer
+/**
+ * Increments internal timer by 1
+ */
+function incrementTimer() {
+  console.log(currentTime);
+  currentTime = parseFloat(currentTime) + 1;
+}
+
+/**
+ * If not playing data and has data start playing
+ */
+function playTimer() {
+  if (runData && !isPlaying) {
+    timer = setInterval(incrementTimer, 100);
+    isPlaying = true;
+    console.log(currentTime);
+  }
+  // if (chartInterval1 || chartInterval2) getDataAtInterval();
+}
+
+/**
+ * Pause timer and set falg to false
+ */
+function pauseTimer() {
+  clearInterval(timer);
+  isPlaying = false;
+  pauseCharts();
+}
+
+/**
+ * Resets timer back to 0
+ */
+function resetTimer() {
+  pauseTimer();
+  currentTime = 0;
+}
+
+// Check Timer for hitting Max
+
+setInterval(() => {

```

Comment:
Why can't you check if you have reached the max when you are incrementing instead of having two parallel timers that are going to be ever so slightly out of sync?